dxr/plugins/xbl/analyzer.py

```diff
@@ -0,0 +1,78 @@
+from warnings import warn
+import xml.parsers.expat as expat
+
+from dxr.indexers import Extent, Position
+from dxr.plugins.xbl.refs import PLUGIN_NAME, TypeRef
+
+
+class XBLAnalyzer(object):
+    def __init__(self, tree, contents, encoding):
+        self.tree = tree
+        self.contents = contents
+        self.lines = contents.splitlines(True)
```

Comment:
I'm starting to wonder whether we should implement a caching method on FileToIndex or FileToSkim that returns the split lines. A lot of plugins do this. I wonder if we'd save any time. Oh well, this is for another time.